src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -376,3 +379,26 @@ export function decodeScalar(
       : undefined;
   return isString(unwrapped) ? unwrapped : undefined;
 }
+
+export function downloadAsCsv(tableData, filename) {
+  const {meta, data} = tableData;
+  if (meta !== undefined) {
+    const headings = Object.keys(meta);
+
+    const csvContent = Papa.unparse({
+      fields: headings,
+      data: data.map(row => {
+        return headings.map(col => disableMacros(row[col]));
+      }),
+    });
+
+    const encodedDataUrl = encodeURI(`data:text/csv;charset=utf8,${csvContent}`);
+
+    // Create a download link then click it, this is so we can get a filename
+    const link = document.createElement('a');
```

Comment:
The other discover uses `window.location.assign()` did that not work here?

Reply:
That works, but you can't change the filename with that method. This way we can include the page title and export time 🕐

Reply:
neat trick

Reply:
This trick goes back to jQuery days. Hacky af. I'm disappointed that they (whoever is responsible for web/HTML standards) haven't improved on this after so many years.